libraries/AP_LTM_Telem/AP_LTM_Telem.cpp

```diff
@@ -0,0 +1,233 @@
+ *      $     T   S   VBAT(mv)  Current(ma)   RSSI  AIRSPEED  ARM/FS/FMOD   CRC
+ * ################################################################################################################# */
+
+#include "AP_LTM_Telem.h"
+
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_Notify/AP_Notify.h>
+#include <GCS_MAVLink/GCS.h>
+#include <AP_RSSI/AP_RSSI.h>
+
+extern const AP_HAL::HAL& hal;
+
+// init - perform required initialisation
+void AP_LTM_Telem::init()
+{
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    
+    _uav.mav_type = gcs().frame_type(); // see MAV_TYPE in Mavlink definition file common.h
+    // check for LTM_Port
+    if ((_port = serial_manager.find_serial(
+            AP_SerialManager::SerialProtocol_LTM_Telem, 0))) {
+        _port->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
+        // initialise UART
+        _port->begin(AP_SERIALMANAGER_LTM_BAUD, AP_SERIALMANAGER_LTM_BUFSIZE_RX,
+                AP_SERIALMANAGER_LTM_BUFSIZE_TX);
+        hal.scheduler->register_io_process(
+                FUNCTOR_BIND_MEMBER(&AP_LTM_Telem::tick, void));
+    }
+}
+
+void AP_LTM_Telem::send_LTM(uint8_t lt_packet[], uint8_t lt_packet_size)
+{
+    // calculate checksum
+    uint8_t LTCrc = 0x00;
+    for (uint8_t i = 3; i < lt_packet_size - 1; i++)
+        LTCrc ^= lt_packet[i];
+    lt_packet[lt_packet_size - 1] = LTCrc;
+    for (uint8_t i = 0; i < lt_packet_size; i++)
+        _port->write(lt_packet[i]);
+}
+
+// GPS frame
+void AP_LTM_Telem::generate_Gframe(void)
+{
+    get_position();
+
+    uint8_t lt_buff[LTM_GFRAME_SIZE];
+    // protocol: START(2 bytes)FRAMEID(1byte)LAT(cm,4 bytes)LON(cm,4bytes)SPEED(m/s,1bytes)ALT(cm,4bytes)SATS(6bits)FIX(2bits)CRC(xor,1byte)
+    // START
+    lt_buff[0] = 0x24; //$
+    lt_buff[1] = 0x54; //T
+    // FRAMEID
+    lt_buff[2] = 0x47; // G ( gps frame at 5hz )
+    // PAYLOAD
+    lt_buff[3] = (_uav.lat >> 8 * 0) & 0xFF;
+    lt_buff[4] = (_uav.lat >> 8 * 1) & 0xFF;
+    lt_buff[5] = (_uav.lat >> 8 * 2) & 0xFF;
+    lt_buff[6] = (_uav.lat >> 8 * 3) & 0xFF;
+    lt_buff[7] = (_uav.lon >> 8 * 0) & 0xFF;
+    lt_buff[8] = (_uav.lon >> 8 * 1) & 0xFF;
+    lt_buff[9] = (_uav.lon >> 8 * 2) & 0xFF;
+    lt_buff[10] = (_uav.lon >> 8 * 3) & 0xFF;
+    lt_buff[11] = (_uav.gndspeed >> 8 * 0) & 0xFF;
+    lt_buff[12] = (_uav.alt >> 8 * 0) & 0xFF;
+    lt_buff[13] = (_uav.alt >> 8 * 1) & 0xFF;
+    lt_buff[14] = (_uav.alt >> 8 * 2) & 0xFF;
+    lt_buff[15] = (_uav.alt >> 8 * 3) & 0xFF;
+    lt_buff[16] = ((_uav.sats_visible << 2) & 0xFF)
+            | ((_uav.fix_type > 3 ? 3 : _uav.fix_type) & 0b00000011); // last 6 bits: sats number, first 2:fix type (0, 1, 2, 3)
+    send_LTM(lt_buff, LTM_GFRAME_SIZE);
+    _ltm_scheduler++;
+}
+
+// Sensors frame
+void AP_LTM_Telem::generate_Sframe(void)
+{
+    get_sense();
+
+    uint8_t lt_buff[LTM_SFRAME_SIZE];

```

Comment:
A static assert for `LTM_FRAME_SIZE` here probably wouldn't be against good taste.

Reply:
Well, I'm not sure what assertion do you mean. Currenlty here are 3 different frame types implemented for LTM protocol in this PR, each one has different size. Any hint? Thx....